src/types.rs

```diff
@@ -434,9 +435,56 @@ impl ObjectFeatures {
     }
 }
 
+/// Information about a Debug Information File in the [`CompleteObjectInfo`].
+///
+/// All DIFs are backed by an [`ObjectFile`](crate::actors::objects::ObjectFile).  But we
+/// may not have been able to get hold of this object file.  We still want to describe the
+/// relevant DIF however.
+///
+/// Currently has no [`ObjectId`] attached and the parent container is expected to know
+/// which ID this DIF info was for.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct DifInfo {
```

Comment:
Can we name this `*Candidate`, and also rename the field to `CompleteObjectInfo::candidates`? 

Also, the other types are usually named around "objects" and not "DIFs". Should we consolidate this one, too?